Userland/Demos/CatDog/CatDog.h

```diff
@@ -25,112 +27,61 @@ class CatDog final : public GUI::Widget
 public:
     static ErrorOr<NonnullRefPtr<CatDog>> create();
 
-    // The general state, does not contain movement direction or whether CatDog is roaming.
-    enum class MainState {
-        Idle,      // default state
-        Alerted,   // woken by mouse cursor or speaking after being idle
-        Sleeping,  // mouse hasn't moved in some time
-        Artist,    // PixelPaint or FontEditor are open
-        Inspector, // SystemServer, Profiler or Inspector are open
-    };
-    static bool is_non_application_state(MainState state)
-    {
-        return state == MainState::Idle || state == MainState::Alerted || state == MainState::Sleeping;
-    }
-
-    virtual void timer_event(Core::TimerEvent&) override;
-    virtual void paint_event(GUI::PaintEvent& event) override;
-    virtual void track_mouse_move(Gfx::IntPoint point) override;
-    virtual void mousedown_event(GUI::MouseEvent& event) override;
-    virtual void context_menu_event(GUI::ContextMenuEvent& event) override;
-
-    void start_the_timer() { m_timer.start(); }
+    void timer_event(Core::TimerEvent&) final;
+    void paint_event(GUI::PaintEvent& event) final;
+    void track_mouse_move(Gfx::IntPoint point) final;
+    void mousedown_event(GUI::MouseEvent& event) final;
+    void context_menu_event(GUI::ContextMenuEvent& event) final;
```

Comment:
Aha, one last thing. Marking these as both `virtual` and `override` was a feature rather than a bug. Rather than following the cpp core guidelines on this one, the preferred style is to mark overridden methods with both keywords.